test/DebugAdapterRunner/DebugAdapterCommand.cs

```diff
@@ -138,12 +151,38 @@ public override void Run(DebugAdapterRunner runner)
             runner.DebugAdapter.StandardInput.Write(request);
 
             // Process + validate responses
-            List<object> responseList = new List<object>();
+            List<ResponsePair> responseList = new List<ResponsePair>();
             int currentExpectedResponseIndex = 0;
+            int previousExpectedResponseIndex = 0;
 
             // Loop until we have received as many expected responses as expected
             while (currentExpectedResponseIndex < this.ExpectedResponses.Count)
             {
+                // Check if previous messages contained the expected response
+                if (previousExpectedResponseIndex != currentExpectedResponseIndex)
+                {
+                    DebugAdapterResponse expected = this.ExpectedResponses[currentExpectedResponseIndex];
+                    foreach (ResponsePair responsePair in responseList)
+                    {
+                        // Make sure we have not seen this response and check to see if it the response we are expecting.
+                        if (!responsePair.Seen && Utils.CompareObjects(expected.Response, responsePair.Response, expected.IgnoreOrder))
```

Comment:
Do we want to add another boolean flag to `DebugAdapterResponse` to be `IngoreResponseOrder` to only allow this for specifc cases / only for `event` types?

Reply:
I think so. Seems like there are cases when we would want to catch cases where we unintentionally re-oredered events.